### DISCUSSION TITLE
[Question][Hw1] meaning of eval function

### DISCUSSION THREAD

**Discussion by Gimongjin** [prosyslab-classroom/cs424-program-reasoning#393] [Q&A]
I'm working on the HW1 and stuck for a long time. I understanded eval function as evaluating the value of instruction and return it as Value.t type.
But I don't have any idea to go forward it. For ConstantInt type, I cannot find the proper function in LLVM OCaml binding document. The function type should be llvalue -> int or similar one I think.
For the Instruction type, I still don't understand how the instruction in eval and instruction in other functions are different, so the former should be interpreted as value and later should be interpreted as State-memory pair. 
Can anyone give me hint for at least part of my question?

**Comment by kohs100** [comment]
There is [Llvm.int64_of_const](https://llvm.moe/ocaml/Llvm.html#VALint64_of_const) function in reference

**Comment by Gimongjin** [comment]
> There is [Llvm.int64_of_const](https://llvm.moe/ocaml/Llvm.html#VALint64_of_const) function in reference

Thanks. Why I couldn't find it?  I will try it.
